xbmc/addons/Repository.cpp

```diff
@@ -55,6 +56,35 @@ using namespace KODI::MESSAGING;
 
 using KODI::MESSAGING::HELPERS::DialogResponse;
 
+CRepository::ResolveResult CRepository::ResolvePathAndHash(const AddonPtr& addon) const
+{
+  CCurlFile file;
+  CURL url{addon->Path()};
+  // Do not follow mirror redirect, we want the headers of the redirect response
+  url.SetProtocolOption("redirect-limit", "0");
+  if (!file.Open(url))
+  {
+    CLog::Log(LOGWARNING, "Could not fetch addon location and hash from {}", url.Get());
+    return {};
+  }
+
+  // Return the location from the header so we don't have to look it up again
+  // (saves one request per addon install)
+  std::string location = file.GetHttpHeader().GetValue("location");
+  // content-* headers are base64, convert to base16
+  std::string hash = StringUtils::ToHexadecimal(Base64::Decode(file.GetHttpHeader().GetValue("content-md5")));
+
+  if (location.empty() || hash.empty())
```

Comment:
Not that I care too much about 3rd party repos, but this will break repos which do not redirect and repos not implementing `content-md5` in the response header.

Reply:
This is on the TODO list ("Only enable the check when it is enabled for the repository").

You should see this PR more as a RFC on the general idea and not dive too deeply into the code yet :-)